x-pack/filebeat/input/cometd/config_auth.go

```diff
@@ -0,0 +1,30 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cometd
+
+import (
+	"errors"
+)
+
+type authConfig struct {
+	OAuth2 *oAuth2Config `config:"oauth2"`
+}
+
+type oAuth2Config struct {
+	// common oauth fields
+	ClientID     string `config:"client.id"`
+	ClientSecret string `config:"client.secret"`
+	User         string `config:"user"`
+	Password     string `config:"password"`
+	TokenURL     string `config:"token_url"`
+}
+
+// Validate checks if oauth2 config is valid.
+func (o *oAuth2Config) Validate() error {
+	if o.TokenURL == "" || o.ClientID == "" || o.ClientSecret == "" || o.User == "" || o.Password == "" {
+		return errors.New("both token_url and client credentials must be provided")
```

Comment:
The error message here doesn't match the conditions being checked. If the `User` is omitted an error referencing `token_url` and `client` will be shown.

Can we make it is so the error message tells the user exactly what fields they need to set?

Reply:
Makes sense, thanks. We'll update it with separate checks.